packages/vuetify/src/components/VDatePicker/VDatePicker.tsx

```diff
@@ -194,6 +194,44 @@ export const VDatePicker = genericComponent<new <
       return targets
     })
 
+    function allowedYears (year: number) {
+      if (typeof props.allowedDates === 'function') {
+        const isAllowedDate = props.allowedDates
+        const start = adapter.parseISO(`${year}-01-01`)
+        const yearSize = adapter.getDiff(adapter.endOfYear(start), start, 'days')
+
+        return createRange(yearSize)
+          .some(count => isAllowedDate(adapter.addDays(start, count)))
+      }
+
+      if (Array.isArray(props.allowedDates) && props.allowedDates.length) {
+        return props.allowedDates.map(date => adapter.getYear(adapter.date(date))).includes(year)
+      }
+
+      return true
+    }
+
+    function allowedMonths (month: number) {
+      if (typeof props.allowedDates === 'function') {
+        const isAllowedDate = props.allowedDates
+        const start = adapter.parseISO(`${year.value}-${month}-01`)
+        const monthSize = adapter.getDiff(adapter.endOfMonth(start), start, 'days')
+
+        return createRange(monthSize)
+          .some(count => isAllowedDate(adapter.addDays(start, count)))
+      }
+
+      if (Array.isArray(props.allowedDates) && props.allowedDates.length) {
+        if (!allowedYears(year.value)) return false
+
+        return props.allowedDates
+          .filter(date => adapter.getYear(adapter.date(date)) === year.value)
+          .map(date => adapter.getMonth(adapter.date(date))).includes(month)
+      }
+
+      return true
+    }
```

Comment:
This looks great. Is it worth making some of the above reusable to avoid duplication?

Reply:
made validation logic for function `allowedDates` reusable